feat: add control to change the kubernetes namespace

deboer-tim opened the pull request:
### What does this PR do?

Creates a new namespace dropdown that shows the currently accessible namespaces
and allows you to change it. This dropdown is added to the Kubernetes Dashboard
and every Kubernetes list page (except Nodes).

### Screenshot / video of UI

https://github.com/user-attachments/assets/95f29481-c1ba-4b01-9dc9-d939bd273a15

### What issues does this PR fix or reference?

Fixes #11274.

### How to test this PR?

Go to the Kubernetes Dashboard and several resource pages, changing the namespace as you go.

- [x] Tests are covering the bug fix or the new feature

axel7083 reviewed (commented):
Provided some feedback in https://github.com/podman-desktop/podman-desktop/issues/11273#issuecomment-2741089447

deboer-tim commented:
Updated to latest design from #11273, code improved to be responsive, tests added, and rebased. Moving to regular review, if there are any final design tweaks I can do another PR.

feloy commented on packages/renderer/src/lib/kube/NamespaceDropdown.svelte:
```diff
@@ -0,0 +1,38 @@
+});
+
+let currentNamespacePromise = $derived.by(async () => {
+  return $kubernetesCurrentContextState.reachable ? await window.kubernetesGetCurrentNamespace() : '';
```
```suggestion
  return $kubernetesCurrentContextState?.reachable ? await window.kubernetesGetCurrentNamespace() : '';
```

I suppose this will fix the problem?

feloy commented:
I can choose the namespace in Kubernetes non-experimental mode, but the control stays gray/disabled in experimental mode. Is it expected?

feloy commented:
> Currently, yes: I added the disabling when disconnected and forgot that in experimental this is different and a bit more complicated. If you want I can work on it here, or follow-up PR.

A follow-up PR is good for me

deboer-tim commented:
@axel7083 @dgolovin could I get a second review?

cdrage reviewed (changes requested):
Unsure of anything changed, but I'm unable to see anything under namespace?

It's just blank.

There's no errors in the devtools console either.


https://github.com/user-attachments/assets/6d1700b5-bbbb-4fc7-bec1-e3bde9231ae8

deboer-tim commented:
Thanks, merging this so that I can rebase the work I started on experimental and get that progressing.

The empty namespace pause is interesting. Since this is reactive I suspect that it's 'correct' and we aren't connected to a namespace for that period, but I can investigate and see where the delay is.
